### ISSUE TITLE
Merge message parser does not work for merge commits created by TFS

### ISSUE THREAD

**Issue by kelly987654** [GitTools/GitVersion#1628]
TFS automatically prepends the PR# to the merge commit message, which prevents the Regex matches of merge commit messages to properly detect the base version.

For ex. if the PR title is 'Merge release/2.3.1 to master', the merge commit automatically created by TFS on completion of the PR is 'Merged PR 12345: Merge release/2.3.1 to master'

Expected version: 2.3.1+0
Actual version: 2.3.1+688

Removing the '^' at the beginning of the Regex expressions in MergeMessage.cs would fix the issue.

This issue appeared after updating to GitVersion 4, we did not have this issue in version 3.

**Comment by kelly987654** [comment]
Any thoughts on this? This issue prevents us from using the latest GitVersion, since we cannot control the merge commit message created by TFS.

**Comment by kelly987654** [comment]
Still hoping for a fix on this

**Comment by asbjornu** [comment]
I'd think #1488 should help fix this issue for you. Have you looked into configuring `merge-message-formats`? Now I just discovered that although #1488 provided documentation, that documentation doesn't seem to have been published on [Read The Docs](https://gitversion.readthedocs.io/en/latest/configuration/). Any idea why that might be, @arturcic?

**Comment by kelly987654** [comment]
Thanks @asbjornu! I will test it out.

**Comment by asbjornu** [comment]
The documentation for `merge-message-formats` is now live on [gitversion.net](https://gitversion.net/docs/configuration#merge-message-formats).

**Comment by kelly987654** [comment]
Thanks - I tried it exactly from the documentation, and it didn't work. I eventually got it working after adding quotes around the regex.

This failed due to an error parsing the YAML:
```
merge-message-formats:
    tfs: ^Merged (?:PR (?<PullRequestNumber>\d+)): Merge (?<SourceBranch>.+) to (?<TargetBranch>.+)
```

This worked:
```
merge-message-formats:
    tfs: '^Merged (?:PR (?<PullRequestNumber>\d+)): Merge (?<SourceBranch>.+) to (?<TargetBranch>.+)'
```

**Comment by kelly987654** [comment]
The original issue seems to be resolved however, thank you for the help.

One other question if you don't mind - we are having issues where users creating the pull request from the release branch to master don't use the exact right syntax (e.g. 'merge release 1.0.0 into master' instead of 'merge release/1.0.0 to master'). And TFS has unfortunately [removed the auto-complete of the merge message title](https://developercommunity.visualstudio.com/content/problem/233063/pull-request-title-not-auto-populating.html). 

Is there any setting or suggestion to designate the commit created by the merge from the release branch into master as the version specified by the release branch name?

**Comment by asbjornu** [comment]
@kelly987654, yes – [YAML](https://yaml.org/) requires strings to be wrapped in quotation marks. The release branch name should affect the version number, as long as it's identified as a release branch by GitVersion. How do you name your release branches? If you use `release/` or `release-` as a prefix, GitVersion should identify it automatically.

**Comment by kelly987654** [comment]
@asbjornu : Thank you!

Our workflow is as follows:
- master: gitversion is 1.1.0
- develop: gitversion is 1.2.0-dev.88
- create new branch from develop named release/1.2.0: gitversion is 1.2.0-rc.0
- merge release/1.2.0 into master:
  - If we don't name the Pull Request title (merge message) exactly right, master branch gitversion is 1.1.1

I tried with GitVersion 5.1.2, as well as the much older version (3.6.5), and it seems that GitVersion only uses the merge message to detect if the release branch was merged into master, and if the merge message isn't formatted correctly, it will default to the branch's default increment (Patch).

Are we missing some other step or configuration that would allow us to be more flexible with the exact merge commit message?
